papermill/translators.py

```diff
@@ -213,12 +213,65 @@ def comment(cls, cmt_str):
         return '# {}'.format(cmt_str).strip()
 
 
+class MatlabTranslator(Translator):
+    @classmethod
+    def translate_escaped_str(cls, str_val):
+        """Translate a string to an escaped Matlab string"""
+        if isinstance(str_val, string_types):
+            str_val = str_val.encode('unicode_escape')
+            if sys.version_info >= (3, 0):
+                str_val = str_val.decode('utf-8')
+            str_val = str_val.replace('"', '""')
+        return '"{}"'.format(str_val)
+
+    @staticmethod
+    def __translate_char_array(str_val):
+        """Translates a string to a Matlab char array"""
+        if isinstance(str_val, string_types):
+            str_val = str_val.encode('unicode_escape')
+            if sys.version_info >= (3, 0):
+                str_val = str_val.decode('utf-8')
+            str_val = str_val.replace('\'', '\'\'')
```

Comment:
This won't catch `\\'` to correctly leave it escaped, but that's probably an edge we don't need to solve for now.

Reply:
Do you have a suggestion how to handle this?

Reply:
If we needed to we would probably do `str_val.replace('\'', '\'\'').replace('\'', '\'\'')` in order to capture escaped string, or use a regex pattern. This is fine thinking about it more, as the original user input shouldn't be doing any pre-escaping patterns for this ever.